src/DataWrangling/ecco_metadata.jl

```diff
@@ -0,0 +1,173 @@
+struct ECCO2Daily end
+struct ECCO4Monthly end
+
+# Metadata holding the ECCO dataset information:
+# - `name`: The name of the dataset.
+# - `dates`: The dates of the dataset, in a `AbstractCFDateTime` format.
+# - `version`: The version of the dataset, could be ECCO2Monthly, ECCO2Daily, or ECCO4Monthly.
+struct ECCOMetadata{D, V} 
+    name  :: Symbol
+    dates :: D
+    version :: V
+end
+
+Base.show(io::IO, metadata::ECCOMetadata) = 
+    print(io, "ECCOMetadata:", '\n',
+    "├── field: $(metadata.name)", '\n',
+    "├── dates: $(metadata.dates)", '\n',
+    "└── data version: $(metadata.version)")
+
+# The default is the ECCO2Daily dataset at 1993-01-01.
+function ECCOMetadata(name::Symbol; 
+                      date = DateTimeProlepticGregorian(1993, 1, 1),
+                   version = ECCO2Daily()) 
+             
+    return ECCOMetadata(name, date, version) 
+end
+
+# Treat ECCOMetadata as an array to allow iteration over the dates.
+Base.getindex(metadata::ECCOMetadata, i::Int) = @inbounds ECCOMetadata(metadata.name, metadata.dates[i], metadata.version)
+Base.length(metadata::ECCOMetadata)       = length(metadata.dates)
+Base.eltype(metadata::ECCOMetadata)       = Base.eltype(metadata.dates)
+Base.first(metadata::ECCOMetadata)        = @inbounds ECCOMetadata(metadata.name, metadata.dates[1], metadata.version)
+Base.last(metadata::ECCOMetadata)         = @inbounds ECCOMetadata(metadata.name, metadata.dates[end], metadata.version)
+Base.iterate(metadata::ECCOMetadata, i=1) = (@inline; (i % UInt) - 1 < length(metadata) ? (@inbounds ECCOMetadata(metadata.name, metadata.dates[i], metadata.version), i + 1) : nothing)
+
+Base.axes(metadata::ECCOMetadata{<:AbstractCFDateTime})    = 1
+Base.first(metadata::ECCOMetadata{<:AbstractCFDateTime})   = metadata
+Base.last(metadata::ECCOMetadata{<:AbstractCFDateTime})    = metadata
+Base.iterate(metadata::ECCOMetadata{<:AbstractCFDateTime}) = (metadata, nothing)
+Base.iterate(::ECCOMetadata{<:AbstractCFDateTime}, ::Any)  = nothing
+
+Base.size(data::ECCOMetadata{<:Any, <:ECCO2Daily})   = (1440, 720, 50, length(data.dates))
+Base.size(data::ECCOMetadata{<:Any, <:ECCO2Monthly}) = (1440, 720, 50, length(data.dates))
+Base.size(data::ECCOMetadata{<:Any, <:ECCO4Monthly}) = (720,  360, 50, length(data.dates))
+
+Base.size(::ECCOMetadata{<:AbstractCFDateTime, <:ECCO2Daily})   = (1440, 720, 50, 1)
+Base.size(::ECCOMetadata{<:AbstractCFDateTime, <:ECCO2Monthly}) = (1440, 720, 50, 1)
+Base.size(::ECCOMetadata{<:AbstractCFDateTime, <:ECCO4Monthly}) = (720,  360, 50, 1)
+
+# The whole range of dates in the different dataset versions
+all_ecco_dates(::ECCO4Monthly) = DateTimeProlepticGregorian(1992, 1, 1) : Month(1) : DateTimeProlepticGregorian(2023, 12, 1)
+all_ecco_dates(::ECCO2Monthly) = DateTimeProlepticGregorian(1992, 1, 1) : Month(1) : DateTimeProlepticGregorian(2023, 12, 1)
+all_ecco_dates(::ECCO2Daily)   = DateTimeProlepticGregorian(1992, 1, 4) : Day(1)   : DateTimeProlepticGregorian(2023, 12, 31)
```

Comment:
for clarify, you probably want to define things like

```julia
const ECCO2_final_time = DateTimeProlepticGregorian(2023, 12, 1)
```

or something. And put them near the definition of the struct `ECCO2Monthly` for example. Because these things will change in the future presumably.